### ISSUE TITLE
mugc script enhancement

### ISSUE THREAD

**Issue by aakifshaikh** [cloud-custodian/cloud-custodian#6681]
**Is your feature request related to a problem? Please describe.**
The current "mugc" script just removes/deletes the Lambda Function Only. It does not remove the log groups and event rules. This lets you write your own script or another set of policies to do that. 

I believe when the custodian policy is deployed it creates 3 things- lambda, log-group, and event-rules - So the cleanup script must remove all three components upon decommissioning. Just deleting the lambda and not other stuff leaves the remnant which becomes grease with overtime. Also, the event-rule quota gets exhausted. 

We were planning to put this into our CICD pipeline because we like the tool and available options and it does what it is supposed to do- but it's doing incomplete cleanup by leaving the log-group and event-rules remnants.

**Comment by ajkerrigan** [comment]
Which sorts of event rules do you see sticking around after a mugc run? In my experience (and looking at [the code](https://github.com/cloud-custodian/cloud-custodian/blob/27752288de5bd8135ef37236db233c580a7b5b26/c7n/mu.py#L405-L409)), events should be removed just before the function gets deleted. If events are sticking around it's worth sorting out why. Policy details and mugc logs should help there.

I'm less sure about logs. I think automatically removing log groups could be an unpleasant surprise, but it does seem like a handy option.

What I've seen more often is [cleaning up stale log groups based on last write time](https://cloudcustodian.io/docs/aws/resources/log-group.html#aws-log-group-actions-delete), which can apply generally or focus on a specific prefix if you want to only target old custodian log groups.

**Comment by aakifshaikh** [comment]
I did a dry run and looks like it is only removing lambda. see screenshot below

![Screen Shot 2021-05-25 at 6 33 50 PM](https://user-images.githubusercontent.com/27826591/119581331-f1ce5d00-bd87-11eb-82b6-350744bc22ca.png)

**Comment by ajkerrigan** [comment]
Ah ok, that makes sense in the context of a dry run. The dry run will show you which functions it'll remove, but doesn't cover the lower level details involved. For the sake of comparison...

Dry run:

```
❯ python ~/code/cloud-custodian/tools/ops/mugc.py -r us-east-2 -v --dryrun --present -c ~/code/sandbox-policies/*.yml
2021-05-25 19:38:20,702: custodian.cache:DEBUG Disabling cache
2021-05-25 19:38:21,753: mugc:INFO Region:us-east-2 Removing custodian-s3-autotag
2021-05-25 19:38:21,753: mugc:INFO Dryrun skipping removal
```

Live run:

```
❯ python ~/code/cloud-custodian/tools/ops/mugc.py -r us-east-2 -v --present -c ~/code/sandbox-policies/*.yml
2021-05-25 19:38:31,079: custodian.cache:DEBUG Disabling cache
2021-05-25 19:38:32,146: mugc:INFO Region:us-east-2 Removing custodian-s3-autotag
2021-05-25 19:38:32,434: custodian.serverless:INFO Removing cwe targets and rule custodian-s3-autotag
2021-05-25 19:38:32,601: custodian.serverless:INFO Removing lambda function custodian-s3-autotag
2021-05-25 19:38:32,775: mugc:INFO Region:us-east-2 Removed custodian-s3-autotag
```

**Comment by aakifshaikh** [comment]
You are right- I just tried mugc in the sandbox and it does delete the cw targets, event-rules and lambda function. And does not touch log-groups.

**Comment by aakifshaikh** [comment]
Do you have any custom script to delete the log-groups. Custodian Policy to delete the log-group does not work.
Here is my use case- If we change the policy name or want to decommission it completely, We want to remove clean-up lambda, cw event-rule, target, and log-group. MUGC solves 70% of the problem, but does not delete the log-group....we want to enhance the script to do that as well.

**Comment by ajkerrigan** [comment]
Just came across this issue again, I don't have any script for that. Though in my experience I've usually wanted to keep the logs around for a while after deleting policies, in which case stale log groups can be cleaned out separately with a policy like:

```yaml
policies:
  - name: delete-stale-custodian-log-groups
    resource: aws.log-group
    filters:
      - type: value
        key: logGroupName
        op: glob
        value: /aws/lambda/custodian-*
      - type: last-write
        days: 90
```

Not sure if that would be useful in your case or not.

**Comment by aakifshaikh** [comment]
@ajkerrigan - thanks for the reply. The policy does not work just because we have some policies that get executed every 3 months, some every 6 months, and annual basis. So deleting the log group with last-write does not work for us. We have created a separate python code to do this internally. I will check if we can share the code....
